subprojects/instant-execution/src/integTest/groovy/org/gradle/instantexecution/InstantExecutionBuildOptionsIntegrationTest.groovy

```diff
@@ -243,4 +243,69 @@ class InstantExecutionBuildOptionsIntegrationTest extends AbstractInstantExecuti
         "isCi.map(String::toBoolean).getOrElse(false)" | "value"
         "isCi.isPresent"                               | "presence"
     }
+
+    @Unroll
+    def "file contents #usage used as build logic input"() {
```

Comment:
perhaps add some coverage that the contents are not treats as build logic input when not queried at configuration time.

We should also look at moving some of the coverage closer to the implementation, for this and the other value sources.

Reply:
Added coverage for that scenario.